core/app/models/spree/promotion_code.rb

```diff
@@ -39,4 +40,8 @@ def usage_limit
   def downcase_value
     self.value = value.downcase
   end
+
+  def remove_excess_whitespace
+    value&.gsub! /\s+/, ''
```

Comment:
It's compat with older Ruby. Solidus currently only requires Ruby 2.2.2 so we can't use the safe navigation operator.